Yafc.Model/Analysis/DependencyNode.cs

```diff
@@ -0,0 +1,296 @@
+    /// <returns>The <see cref="Bits"/> for all the dependent items, ORed together appropriately.</returns>
+    internal abstract Bits AggregateBits(Func<FactorioId, Bits> getBits);
+
+    /// <summary>
+    /// Gets the new <see cref="AutomationStatus"/> for this tree's owner, based on the current value and the statuses returned by
+    /// <paramref name="isAutomatable"/>.
+    /// </summary>
+    /// <param name="isAutomatable">A delegate that returns the <see cref="AutomationStatus"/> for a given <see cref="FactorioId"/>.</param>
+    /// <param name="automationState">The initial <see cref="AutomationStatus"/>, which will not be exceeded regardless of the values returned
+    /// by <paramref name="isAutomatable"/>.</param>
+    /// <returns>The highest feasible <paramref name="automationState"/>, as constrained by <paramref name="isAutomatable"/> and
+    /// <paramref name="automationState"/>.</returns>
+    internal abstract AutomationStatus IsAutomatable(Func<FactorioId, AutomationStatus> isAutomatable, AutomationStatus automationState);
+
+    /// <summary>
+    /// Instructs this dependency tree to draw itself on the specified <see cref="ImGui"/>.
+    /// </summary>
+    /// <param name="gui">The drawing destination.</param>
+    /// <param name="builder">A delegate that will draw the passed <see cref="DependencyList"/> onto the passed <see cref="ImGui"/>.</param>
+    public abstract void Draw(ImGui gui, Action<ImGui, DependencyList> builder);
+
+    /// <summary>
+    /// A <see cref="DependencyNode"/> that requires all of its children.
+    /// </summary>
+    private sealed class AndNode : DependencyNode {
+        private readonly DependencyNode[] dependencies;
+
+        private AndNode(DependencyNode[] dependencies) => this.dependencies = dependencies; // Use Create
+
+        /// <summary>
+        /// Returns a <see cref="DependencyNode"/> that requires all of the children specified in <paramref name="dependencies"/>.
+        /// </summary>
+        /// <param name="dependencies">The children that should all be required by the returned value.</param>
+        internal static DependencyNode Create(IEnumerable<DependencyNode> dependencies) {
+            if (dependencies.Count() == 1) {
+                return dependencies.First();
+            }
+
+            List<DependencyNode> realDependencies = [];
+            foreach (DependencyNode item in dependencies) {
+                if (item is AndNode and) {
+                    realDependencies.AddRange(and.dependencies);
+                }
+                else {
+                    realDependencies.Add(item);
+                }
+            }
+            realDependencies = realDependencies.Distinct().ToList();
+            if (realDependencies.Count == 1) {
+                return realDependencies[0];
+            }
+            return new AndNode([.. realDependencies]);
```

Comment:
We could have an `AndNode` with just one dependency. It would (technically) work, and it allows to get rid of the 'if one dependency then special case' check/code

Reply:
Kept to keep unpacking code less complex